Knowledge-Base/Virtiofs-qs.md

```diff
@@ -0,0 +1,255 @@
+   Virtio-FS requires memory sharing between host and guest.
+
+   - Go to the **Memory** tab.
+   - Check the box **Enable shared memory**.
+
+   > If this option is not available in the GUI, you can add it manually in the domain XML:
+   >
+   > ```xml
+   > <memoryBacking>
+   >   <source type='memfd'/>
+   >   <access mode='shared'/>
+   > </memoryBacking>
+   > ```
+
+2. **Add Virtio-FS Filesystem Device**  
+   - Go to **Add Hardware** → **File System**
+   - Choose:
+     - **Driver**: `virtiofs`
+     - **Source Path**: the host directory you want to share (e.g. `/home/user/viofs`)
+     - **Target Path**: a mount tag (e.g. `mount_tag0`) – this is used by the guest to identify the shared file system.
+
+3. **Apply and Start the VM**  
+   - Save your configuration.
+   - Start the VM.  
+   - In the guest use the matching `mount_tag0` in the mount command.  
+
+
+More information on libvirt virtiofs options is provided in [libvirt docs](https://libvirt.org/kbase/virtiofs.html).
+
+### QEMU
+
+#### 1. Start virtiofsd daemon [(install or build it if missing)](#virtiofsd):
+
+```
+/usr/libexec/virtiofsd --socket-path=/tmp/virtiofs_socket -o source=/home/user/viofs
+```
+
+The socket path will also need to be passed to QEMU below.  
+#### 2. Adjust following QEMU command-line parameters to your QEMU command line:
+
+* Instantiate the character device for socket communication between QEMU and virtiofsd:  
+`-chardev socket,id=char0,path=/tmp/virtiofs_socket`
+
+* Instantiate the virtiofs PCI device:  
+`-device vhost-user-fs-pci,queue-size=1024,chardev=char0,tag=my_virtiofs`  
+The tag name is arbitrary and must match the tag given in the guests mount command.
+
+
+* Force use of memory sharing with virtiofsd (replace `4G` with your desired VM RAM size):  
+`-m 4G -object memory-backend-file,id=mem,size=4G,mem-path=/dev/shm,share=on -numa node,memdev=mem`
+
+#### QEMU Command with Virtio-FS Example
+```
+qemu-system-x86_64 \
+  -enable-kvm \
+  -m 4G \
+  -object memory-backend-file,id=mem,size=4G,mem-path=/dev/shm,share=on -numa node,memdev=mem \
+  -chardev socket,id=char0,path=/tmp/virtiofs_socket \
+  -device vhost-user-fs-pci,queue-size=1024,chardev=char0,tag=my_virtiofs \
+  ...
+```
+
+### virtiofsd
+
+The `virtiofsd` daemon is required on the host to enable Virtio-FS for guests.
+
+* #### Installation (Recommended)
+  On `Fedora` virtiofsd is available via the package manager.  
+  ```
+  sudo dnf install virtiofsd
+  ```
+
+* #### Building from Source (Alternative)
+  Modern virtiofsd is written in Rust and supported here: https://gitlab.com/virtio-fs/virtiofsd.  
+  To build virtiofsd:
+  ```
+  cargo build
+  ```
+
+
+## Guest
+
+
+### Setup with installer
+1. Download and install [WinFSP](https://github.com/billziss-gh/winfsp/releases) with at least "Core" feature enabled.
+2. Install virtiofs driver and service from [VirtIO-Win package](https://github.com/virtio-win/virtio-win-pkg-scripts/blob/master/README.md).
```

Comment:
I suggest using this URL instead: https://github.com/virtio-win/virtio-win.github.io/blob/main/Knowledge-Base/Driver-installation.md#variant-1-using-the-installation-wizard

@YanVugenfirer What do you think?

Reply:
Yes, this one better